### PR TITLE
Feature/hosiptal registration in pharmacy

### PR DESCRIPTION
**Pull request by tamaragovorcin**

### CODE REVIEW

File: IntegrationWithPharmacies/Controllers/VueHelper.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                var timeout = Task.Delay(Timeout);
+                if (await Task.WhenAny(timeout, tcs.Task) == timeout)
+                {
+                    throw new TimeoutException();
+                }
+
+                return DevelopmentServerEndpoint;
+            });
+
+        }
```
Comment (Maja0505) [inline IntegrationWithPharmacies/Controllers/VueHelper.cs:95]: It is necessary to refactor this functionality

File: IntegrationWithPharmacies/Controllers/RegistrationController.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        public RegistrationController(MyDbContext context)
+        {
+            registrationInPharmacyService = new RegistrationInPharmacyService(context);
+        }
+
+        [HttpGet]   // GET /api/registration
+        public IActionResult Get()
+        {
+            List<RegistrationInPharmacy> result = registrationInPharmacyService.GetAll();
+            return Ok(result);
```
Comment (Nemanja0505) [inline IntegrationWithPharmacies/Controllers/RegistrationController.cs:32]: Unnecessary instance of List<RegistrationInPharmacy>, you should return it directly through the Ok() method.
return Ok(registrationInPharmacyService.GetAll())

File: IntegrationWithPharmacies/Controllers/RegistrationController.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        }
+
+        [HttpPost]      // POST /api/registration Request body: {"pharmacyId": "Some number", "apiKey": "Some api key"}
+        public IActionResult Create(RegistrationInPharmacyDto dto)
+        {
+            RegistrationInPharmacy registration = registrationInPharmacyService.Create(dto);
+
+            if (registration == null) return BadRequest();
+
+            return Ok();
```
Comment (Nemanja0505) [inline IntegrationWithPharmacies/Controllers/RegistrationController.cs:42]: Unnecessary instance of object, use it directly in the if statement.

File: IntegrationWithPharmacies/Startup.cs [hunk 4]
```diff
@@ -24,7 +27,24 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
+            services.AddControllersWithViews().AddNewtonsoftJson(options =>
+               options.SerializerSettings.ReferenceLoopHandling = Newtonsoft.Json.ReferenceLoopHandling.Ignore);
+
             services.AddControllers();
+            services.AddDbContext<MyDbContext>(options =>
+            options.UseMySql(ConfigurationExtensions.GetConnectionString(Configuration, "MyDbContextConnectionString")).UseLazyLoadingProxies());
```
Comment (Maja0505) [inline IntegrationWithPharmacies/Startup.cs:35]: Use same form as in 30. and 31.  lines

File: IntegrationWithPharmacies/Startup.cs [hunk 5]
```diff
@@ -24,7 +27,24 @@ public Startup(IConfiguration configuration)
...
+            {
+                options.AddPolicy("VueCorsPolicy", builder =>
+                {
+                    builder
+                      .AllowAnyHeader()
+                      .AllowAnyMethod()
+                      .AllowCredentials()
+                      .WithOrigins("http://localhost:57942");
+                });
+            });
```
Comment (Maja0505) [inline IntegrationWithPharmacies/Startup.cs:47]: Use same form as in 30. and 31.  lines

File: PatientWebApplication/HealthClinic.DAL/Adapters/RegistrationInPharmacyAdapter.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+using System.Text;
+
+namespace HealthClinic.CL.Adapters
+{
+    public class RegistrationInPharmacyAdapter
+    {
+        public static RegistrationInPharmacy RegistrationDtoToRegistration(RegistrationInPharmacyDto dto)
+        {
+            RegistrationInPharmacy registration = new RegistrationInPharmacy(1, dto.pharmacyId, dto.apiKey);
+            return registration;
```
Comment (Nemanja0505) [inline PatientWebApplication/HealthClinic.DAL/Adapters/RegistrationInPharmacyAdapter.cs:14]: Unnecessary instance of object, return it directly.

File: PatientWebApplication/HealthClinic.DAL/Adapters/RegistrationInPharmacyAdapter.cs [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+        }
+
+        public static RegistrationInPharmacyDto RegistrationToRegistrationDto(RegistrationInPharmacy registration)
+        {
+            RegistrationInPharmacyDto dto = new RegistrationInPharmacyDto(registration.pharmacyId, registration.apiKey);
+            return dto;
```
Comment (Nemanja0505) [inline PatientWebApplication/HealthClinic.DAL/Adapters/RegistrationInPharmacyAdapter.cs:20]: Unnecessary instance of object, return it directly.

**Review by Nemanja0505 — CHANGES_REQUESTED** [review]
Fix commented parts of code.

**Review by Maja0505 — CHANGES_REQUESTED** [review]
Fix commented parts of code

**Review by Maja0505 — APPROVED** [review]
PR is approved. The task solution is implemented with good Clean code principles.

**Review by Nemanja0505 — APPROVED** [review]
PR is approved. The task solution is implemented with good Clean code principles.

**Pull request merged**